### PR TITLE
Add Silo and Ditch PROTO models

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Ditto.

### CODE REVIEW

File: projects/objects/buildings/protos/Silo.proto [hunk 1]
```diff
@@ -0,0 +1,170 @@
+#VRML_SIM R2023b utf8
+# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+# license url: https://cyberbotics.com/webots_assets_license
+# documentation url: https://webots.cloud/run?url=https://github.com/cyberbotics/webots/blob/released/projects/objects/buildings/protos/Silo.proto
+# keywords: building/farm
+# A steel grain silo.
+# template language: javascript
+
+EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BrushedSteel.proto"
```
Comment (BenjaminDeleze) [inline projects/objects/buildings/protos/Silo.proto:9]:
```suggestion
EXTERNPROTO "webots://projects/appearances/protos/BrushedSteel.proto"
```

File: projects/objects/buildings/protos/Silo.proto [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+PROTO Silo [
+  field SFVec3f    translation          0 0 0
+  field SFRotation rotation             0 0 1 0
+  field SFFloat    height               17
+  field SFFloat    radius               3
+  field SFString   name                 "silo"
+  field SFBool     enableBoundingObject TRUE    
+  field SFBool     locked               TRUE
```
Comment (BenjaminDeleze) [inline projects/objects/buildings/protos/Silo.proto:18]: Is it really a good idea to lock it by default?
As a user I would like to insert it, move it and then lock it.

File: projects/objects/floors/protos/Ditch.proto [hunk 3]
```diff
@@ -0,0 +1,170 @@
+#VRML_SIM R2023b utf8
+# license: Copyright Cyberbotics Ltd. Licensed for use only with Webots.
+# license url: https://cyberbotics.com/webots_assets_license
+# documentation url: https://webots.cloud/run?url=https://github.com/cyberbotics/webots/blob/released/projects/objects/floors/protos/Ditch.proto
+# tags: nonDeterministic
+# keywords: primitive/ground
+# Randomly generated ditch with optional water.
+# template language: javascript
+
+EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/SandyGround.proto"
```
Comment (BenjaminDeleze) [inline projects/objects/floors/protos/Ditch.proto:10]:
```suggestion
EXTERNPROTO "webots://projects/appearances/protos/SandyGround.proto"
```

File: projects/objects/floors/protos/Ditch.proto [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+PROTO Ditch [
+  field SFVec3f    translation    0 0 0
+  field SFRotation rotation       0 0 1 0
+  field SFString   name           "ditch"
+  field SFVec3f    size           20 5 2     # Defines the size of the ditch.
+  field SFFloat    waterHeight    0.3        # Defines the height of the water in the ditch.
+  field SFNode     appearance     SandyGround { textureTransform TextureTransform { scale 5 5 } }  # Defines the appearance of the terrain.
+  field SFInt32    randomSeed     1                                                                  # Defines whether the bounds of the terrain should be flat.
```
Comment (BenjaminDeleze) [inline projects/objects/floors/protos/Ditch.proto:19]: The alignment of comments seems wrong

**Review by BenjaminDeleze — CHANGES_REQUESTED** [review]
Thank you!
Just some minor remarks:
- It would be nice to integrate the new protos in existing world (in `Boomer.wbt` for example).
- The appearance of the `Silo` does not really match with its icon (in which the Silo is green). And I think that the current appearance of the Silo is too `metallic`, but that's subjective.
- As `coneHeight` is already a variable, it may be nice to make it a parameter as 2 meters could be strange depending on the `radius`.

**Comment by stefaniapedrazzi** [comment]
> The appearance of the Silo does not really match with its icon (in which the Silo is green). 

It is metallic, so it reflects the background and the appearance depends on the background used in the world.

**Comment by stefaniapedrazzi** [comment]
The conflicts has been resolved, but this GitHub PR page doesn't get updated.
I will wait a little bit to check if it changes otherwise I will close this PR and open a new one.

**Pull request closed without merging**